data/binding/mapbinding.go

```diff
@@ -250,20 +250,19 @@ func (b *mapBase) doReload() (retErr error) {
 		b.trigger()
 	}
 
+	var err error
 	for k, item := range b.items {
-		var err error
-
 		if b.updateExternal {
-			err = item.(*boundExternalMapValue).setIfChanged((*b.val)[k])
-		} else {
-			err = item.(*boundMapValue).set((*b.val)[k])
+			if r := item.(*boundExternalMapValue).setIfChanged((*b.val)[k]); r != nil {
+				err = r
+			}
+			continue
 		}
-
-		if err != nil {
-			retErr = err
+		if r := item.(*boundMapValue).set((*b.val)[k]); r != nil {
+			err = r
 		}
 	}
-	return
+	return err
```

Comment:
It is considered best practice in Go. For example https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
The return value of `updateExternal` is not one that indicates happy path or not, there are two possible, equally weighted, outcomes.
If, instead, you use a measure of cyclomatic complexity the replacement code is 1 order more complex.

Either way we don't typically use `continue` instead of `else` - we do like early exits and left aligned happy path  but neither seem to fit in this case.

Reply:
With this change I was trying to adhere to 2 "standards":
1) your own standard of keeping the error scope small by using `if err := someFunc(); err != nil` syntax
2) using the above mentioned general Go standard of early `return` and early `continue` to keep code easier to read.

I can undo the change if you don't want it here.

Reply:
Moving the `err` and changing the way the return value is handled looks good.
HoweverI would prefer that the inside of the `if updateExternal` blocks return to if/else as in this case it shows the balanced weighting of the loop performing two options per item. The replaced code makes this less clear.